### PR TITLE
LVNOD-27_Сreated_model_of_the-user

### PR DESCRIPTION
**Pull request by yarvanya**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-27)

## Code reviewers

- [ ] @yurii-kokhalevych
- [ ] @shyyka
- [ ] @yuliannakorol

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

First, I have created the model and migration.
Then I've created seed file for user model. After that all needed controllers with its routes.

## CHECK LIST

- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: server/controllers/users.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const User = require('../../config/db').Users;
+const Profile = require('../../config/db').Profiles;
+
+module.exports = {
+  create(req, res) {
+    return User
+      .create({
+        email: req.body.email,
```
Comment (omaik) [inline server/controllers/users.js]: I think you can do this in better way

**Review by shyyka — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

File: server/migrations/20170524175710-create-user.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      },
+      is_activate: {
+        type: Sequelize.BOOLEAN,
+        allowNull: false
+      },
+      status_state: {
+        type: Sequelize.STRING,
+        allowNull: false
+      },
+      created_at: {
```
Comment (omaik) [inline server/migrations/20170524175710-create-user.js]: do you have something like `timestamps` ? it creates both `created_at` and `updated_at` fields

File: server/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    return User
+      .findById(req.params.id, {})
+      .then(user => {
+        if (!user) {
+          return res.status(404).send({
+            message: 'User Not Found',
+          });
+        }
+        return user
+          .update({
```
Comment (4asof4ik) [inline server/controllers/users.js]: and here

File: server/controllers/users.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      })
+      .catch((error) => res.status(400).send(error));
+  },
+
+  destroy(req, res) {
+    return User
+      .findById(req.params.id)
+      .then(user => {
+        if (!user) {
+          return res.status(400).send({
```
Comment (4asof4ik) [inline server/controllers/users.js]: status should be 404 if not  found

File: server/controllers/users.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+'use strict';
```
Comment (4asof4ik) [inline server/controllers/users.js]: rename file

File: server/controllers/users.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          ['created_at', 'ASC']
+        ],
+      })
+      .then((users) => res.status(200).send(users))
+      .catch((error) => res.status(400).send(error));
+  },
+
+  retrieve(req, res) {
+    return User
+      .findById(req.params.id, {})
```
Comment (4asof4ik) [inline server/controllers/users.js]: empty object???

File: server/controllers/users.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      .then((user) => {
```
Comment (4asof4ik) [inline server/controllers/users.js]: if one argument drop () around a user

File: server/routes/index.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const userController = require('../controllers/users');
+
+module.exports = (app) => {
+  app.get('/api', (req, res) => res.status(200).send({
+    message: 'Welcome to the Da - Mi project !!!',
```
Comment (4asof4ik) [inline server/routes/index.js]: drop comma in the end of line

File: server/routes/index.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+const userController = require('../controllers/users');
+
+module.exports = (app) => {
```
Comment (4asof4ik) [inline server/routes/index.js]: ()

File: server/models/user.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      allowNull: false,
+      validate: {
+        isEmail: true
+      }
+    },
+    password: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+      validate: {
+        len: [6,20]
```
Comment (4asof4ik) [inline server/models/user.js]: add validation for a-zA-Z 0-9 and !@#

File: server/controllers/users.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+          return res.status(400).send({
+            message: 'User has not found. Please try again!',
+          });
+        }
+        return user
+          .destroy()
+          .then(() => res.status(204).send())
+          .catch((error) => res.status(400).send(error));
+      })
+      .catch((error) => res.status(400).send(error));
```
Comment (4asof4ik) [inline server/controllers/users.js]: () check all code

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Pull request closed without merging**